diesel/src/pg/query_builder/copy/mod.rs

```diff
@@ -0,0 +1,158 @@
+use crate::pg::Pg;
+use crate::query_builder::nodes::StaticQueryFragment;
+use crate::query_builder::ColumnList;
+use crate::query_builder::QueryFragment;
+use crate::sql_types::SqlType;
+use crate::Expression;
+use crate::{Column, Table};
+
+pub(crate) mod copy_in;
+pub(crate) mod copy_out;
+
+pub(crate) use self::copy_in::{CopyInExpression, InternalCopyInQuery};
+pub(crate) use self::copy_out::CopyTo;
+
+pub use self::copy_in::CopyHeader;
+
+const COPY_MAGIC_HEADER: [u8; 11] = [
+    0x50, 0x47, 0x43, 0x4F, 0x50, 0x59, 0x0A, 0xFF, 0x0D, 0x0A, 0x00,
+];
+
+#[derive(Default, Debug, Copy, Clone)]
+pub enum CopyFormat {
+    #[default]
+    Text,
+    Csv,
+    Binary,
+}
+
+impl CopyFormat {
+    fn to_sql_format(&self) -> &'static str {
+        match self {
+            CopyFormat::Text => "text",
+            CopyFormat::Csv => "csv",
+            CopyFormat::Binary => "binary",
+        }
+    }
+}
+
+#[derive(Default, Debug)]
+struct CommonOptions {
```

Comment:
`ENCODING` is currently always assumed to be the same as for the connection (so `UTF-8`). It might be worth to expose that at some point.
`HEADER` is currently only exposed for `copy_in`, but after reading the documentation again it seems to be useful to expose it also for `copy_out` without allowing `MATCH` 
`FORCE_QUOTE` support requires figuring out passing down a list of columns to that function. That's not impossible, it's just something that I did not want to do for the first version.